packages/evm-adapter/src/getEthPaymentTx.ts

```diff
@@ -14,18 +18,22 @@ export const getEthPaymentTx = async (
   if (!contractAddress) {
     throw new Error(`Non existant contract address for chainId ${chainId}`);
   }
-  const contract = new Contract(contractAddress, helio.abi, provider);
-  const unsignedTx = await contract.populateTransaction.ethPayment(
+  const contract = new Contract(contractAddress, getAbi(chainId), provider);
+  const feesAndAddresses = getFeesAndAddresses(req);
+  const params: Array<any> = [
     req.recipientAddress,
     BigNumber.from(req.amount),
     BigNumber.from(req.fee),
     req.transactonDbId,
-    {
-      value: BigNumber.from(req.amount),
-      gasLimit,
-      gasPrice: await provider.getGasPrice(),
-    }
-  );
+  ];
+  if (isPolygon(chainId)) {
+    params.push(feesAndAddresses);
+  }
```

Comment:
this file is getEthPaymentTx, should never be polygon here?
(and reverse in getPaymenTx i believe)